.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,150 @@
+{
+  "name": "mne-dev",
+  // "image": "mcr.microsoft.com/devcontainers/base:debian-11",
+  "build": { "dockerfile": "Dockerfile" },
+  "containerEnv": {
+    "PYTHONNOUSERSITE": "true", // Make Python ignore the user's site-packages folder if it exists
+    "XDG_RUNTIME_DIR": "/home/vscode/.cache/xdgr"
+  },
+  "features": {
+    // See https://containers.dev/features
+    // APT packages
+    "ghcr.io/rocker-org/devcontainer-features/apt-packages:1": {
+      "packages": "mesa-utils,libegl1,^libxcb.*-dev,libx11-xcb-dev,libglu1-mesa-dev,libxrender-dev,libxi-dev,libxkbcommon-dev,libxkbcommon-x11-dev"
+    },
+    // Git
+    "ghcr.io/devcontainers/features/git:1": {
+      "version": "os-provided",
+      "ppa": "true"
+    },
+    // Conda and Mamba
+    "ghcr.io/mamba-org/devcontainer-features/micromamba:1": {
+      "channels": "conda-forge",
+      "packages": "conda mamba ipython jupyterlab mne"
+    },
+    // Docker interface with host's Docker service
+    "ghcr.io/devcontainers/features/docker-outside-of-docker:1": {},
+    // Zsh plugins
+    "ghcr.io/devcontainers-contrib/features/zsh-plugins:0": {
+      "plugins": "zsh-autosuggestions zsh-syntax-highlighting history-substring-search",
+      "omzPlugins": "https://github.com/zsh-users/zsh-autosuggestions https://github.com/zsh-users/zsh-syntax-highlighting"
+    },
+    // Desktop and VNC access
+    "ghcr.io/devcontainers/features/desktop-lite:1": {}
+  },
+  // Configure tool-specific properties.
+  "customizations": {
+    // Configure properties specific to VS Code.
+    "vscode": {
+      "settings": {
```

Comment:
seems like fertile ground for bikeshedding. Is there a way for the container to pick up the user's local settings?  If not, then can you indicate which of these settings are truly *necessary* for the container to work, which are not necessary but clearly a good idea (e.g., don't restore port forwards?), and which are your *preferences* (perhaps "format on type" fits here)?